src/instructlab/rag/haystack/components/document_splitter.py

```diff
@@ -0,0 +1,102 @@
+# Standard
+from typing import Any, Dict, List, cast
+import logging
+
+# Third Party
+from docling_core.transforms.chunker.hybrid_chunker import HybridChunker
+from docling_core.types import DoclingDocument
+from docling_core.types.legacy_doc.document import (
+    ExportedCCSDocument as LegacyDoclingDocument,
+)
+from docling_core.utils.legacy import legacy_to_docling_document
+from haystack import Document, component  # type: ignore
+from haystack.core.serialization import (  # type: ignore
+    default_from_dict,
+    default_to_dict,
+)
+from pydantic_core._pydantic_core import ValidationError
+
+logger = logging.getLogger(__name__)
+
+
+@component
+class DoclingDocumentSplitter:
+    SUPPORTED_CONTENT_FORMATS = ["json"]
+
+    def __init__(self, embedding_model_id=None, content_format=None, max_tokens=None):
+        self.__chunker = HybridChunker(
+            tokenizer=embedding_model_id, max_tokens=max_tokens
+        )
+        self.__embedding_model_id = embedding_model_id
+
+        if content_format not in self.SUPPORTED_CONTENT_FORMATS:
+            raise ValueError(
+                f"Only the following input formats are currently supported: {self.SUPPORTED_CONTENT_FORMATS}."
+            )
+        self.__content_format = content_format
+
+    @component.output_types(documents=List[Document])
+    def run(self, documents: List[Document]):
+        if not isinstance(documents, list) or (
+            documents and not isinstance(documents[0], Document)
+        ):
+            raise TypeError(
+                "DoclingDocumentSplitter expects a List of Documents as input."
+            )
```

Comment:
Are the checks/exceptions in this class reproducing existing docling checks/exceptions? If so, we can just let it throw.

Reply:
Actually these are the Haystack errors raised during input parameter validation, I don't think we can rely on Docling for that. Any possible exception raised by Docling will be just thrown back to the caller.

Please note that the [docling-haystack](https://github.com/DS4SD/docling-haystack/tree/main) package may probably define its own "official" version of this Haystack component, so we may dismiss ours at that time.

Reply:
Raised [issue #4](https://github.com/DS4SD/docling-haystack/issues/4) to request the component to be defined.